subsystems/shootingsubsystem.py

```diff
@@ -0,0 +1,165 @@
+            constants.kHoodMotorId,
+            constants.kSimHoodMotorPort,
+            constants.kHoodPGain,
+            constants.kHoodIGain,
+            constants.kHoodDGain,
+            constants.kHoodPIDSlot,
+        )
+
+        self.initializationMinimum = 0
+        self.initializationMaximum = 0
+
+        self.targetTurretAngle = Rotation2d()
+        self.targetHoodAngle = Rotation2d()
+        self.targetWheelSpeed = 0
+
+    def setAsStartingPosition(self) -> None:
+        self.hoodMotor.setCurrentEncoderPulseCount(constants.kHoodStartingAngle)
+        if (
+            RobotBase.isReal()
+        ):  # only possible to calibrate the real robot, sim is perfect by default
+            turretRealPosition = map_range(
+                self.turretMotor.getPosition(),
+                self.initializationMinimum,
+                self.initializationMaximum,
+                constants.kTurretMinimum.radians()
+                * constants.kTalonEncoderPulsesPerRadian,
+                constants.kTurretMaximum.radians()
+                * constants.kTalonEncoderPulsesPerRadian,
+            )
+            self.turretMotor.setCurrentEncoderPulseCount(turretRealPosition)
+
+    def periodic(self) -> None:
+        self.initializationMinimum = min(
+            self.initializationMinimum, self.turretMotor.getPosition()
+        )
+        self.initializationMaximum = max(
+            self.initializationMaximum, self.turretMotor.getPosition()
+        )
+        if not SmartDashboard.getBoolean(constants.kShootingManualModeKey, False):
+            SmartDashboard.putNumber(
+                constants.kShootingWheelSpeedKey, self.getWheelSpeed()
+            )
+            SmartDashboard.putNumber(
+                constants.kShootingHoodAngleKey, self.getHoodAngle().degrees()
+            )
+            SmartDashboard.putNumber(
+                constants.kShootingTurretAngleKey, self.getTurretRotation().degrees()
+            )
+        SmartDashboard.putBoolean(constants.kShootingOnTargetKey, self.onTarget())
+
+    def onTarget(self) -> bool:
+        if (
+            abs(self.getWheelSpeed() - self.targetWheelSpeed)
+            <= constants.kWheelSpeedTolerence
+            and abs((self.getHoodAngle() - self.targetHoodAngle).radians())
+            <= constants.kHoodAngleTolerence.radians()
+            and abs((self.getTurretRotation() - self.targetTurretAngle).radians())
+            <= constants.kTurretAngleTolerence.radians()
+        ):
+            return True
+        return False
+
+    def getWheelSpeed(self) -> int:
+        """returns wheel speed in RPM"""
+        return self.shootingMotor.getSpeed()
+
+    def setWheelSpeed(self, speed: int) -> None:
+        self.targetWheelSpeed = speed
+        self.shootingMotor.setSpeed(speed)
+
+    def setHoodAngle(self, angle: Rotation2d) -> None:
+        """angle to fire the ball with
+        absolute with 0 being straight and 90 degrees being direct to the sky"""
```

Comment:
Is this actually true? Yesterday when testing it was about 0 to 15.7 for furthest shot

Reply:
Yeah the commend needs updating, 0 is full upwards